client/app/components/parameters.js

```diff
@@ -77,21 +34,26 @@ function ParametersDirective($location, $uibModal) {
         );
       }
 
-      scope.showParameterSettings = (param) => {
-        $uibModal.open({
-          component: 'parameterSettings',
-          resolve: {
-            parameter: param,
-          },
-        });
+      scope.showParameterSettings = (parameter, index) => {
+        EditParameterSettingsDialog
+          .showModal({ parameter })
+          .result.then((updated) => {
+            // update parameters
+            scope.parameters[index] = extend(parameter, updated);
+
+            // save if already dirty
+            // https://discuss.redash.io/t/query-unsaved-changes-indication/3302/5
+            if (!scope.queryIsDirty) {
```

Comment:
Maybe this component should receive an `onUpdate` callback and handle the logic of saving the query in the query component?

Reply:
Thought of that, but then figured that this code:
```js
.result.then((updated) => {
  // update parameters
  scope.parameters[index] = extend(parameter, updated);
  scope.updateQuery();
}
```
might mislead developers, who won't understand why sometimes it saves and sometimes doesn't. wdyt?

Reply:
Call it `onUpdate` and then it's clear that there is more logic in the handler, no?

Reply:
Update in https://github.com/getredash/redash/pull/3488/commits/c9ab9e173b93a736285a29ec915d27083d69a888